client/blocks/comment-detail/comment-detail-header.jsx

```diff
@@ -28,7 +28,7 @@ import { gmtOffset, timezone } from 'lib/site/utils';
 const getRelativeTimePeriod = ( commentDate, site, moment ) => {
 	const localizedDate = convertDateToUserLocation(
 		commentDate || moment(),
-		timezone( site ),
+		moment.tz.guess() || timezone( site ),
```

Comment:
I feel like I need a unit test here to understand the cases or a bit more clarification. The user's preferred time zone can differ from the site's.

Let's say my local timezone is UTC−08:00, but my site timezone is set to UTC-5:00. What happens here? Should we always be trying to pass through the local user's timezone?

Reply:
Apparently we are not storing the user's timezone. We have a site setting that can be either a timezone string or a GMT offset. And indeed, the real user's timezone can differ from the site's, unless correctly configured. The default site setting is `UTC-0`.

I've removed all the checks because moment's `fromNow` does the conversion for us.

Reply:
Thanks for the clarification @rodrigoi!